components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -302,7 +302,14 @@ class GeckoEngineSession(
                     is InterceptionResponse.Url -> loadUrl(url)
                 }
             }
-            return GeckoResult.fromValue(if (response != null) AllowOrDeny.DENY else AllowOrDeny.ALLOW)
+
+            return if (response != null) {
+                GeckoResult.fromValue(AllowOrDeny.DENY)
+            } else {
+                notifyObservers { onLoadRequest(request.isRedirect) }
```

Comment:
@pocmo this part I don't understand. Why does `request.isRedirect` mean `triggeredByUserInteraction`.

Reply:
Yeah, I didn't understand this either. But I verified that that's what it actually does (and it says so in the javadoc) and that's how it's used in Focus too. And then filed this yesterday:
https://bugzilla.mozilla.org/show_bug.cgi?id=1545170

Reply:
I'll add this as a comment here (with link to the bug)

Reply:
👍